src/detect.sh

```diff
@@ -106,7 +106,12 @@ echo "NSIMD_SIMD           = ${NSIMD_SIMD}"
 echo "NSIMD_OPTIONALS      = ${NSIMD_OPTIONALS}"
 echo "END MAKE_DEFINITION"
 
-# Pass options to Cactus
+# Pass compiler options to Cactus
+echo "BEGIN DEFINE"
+echo "NSIMD_${NSIMD_SIMD}"
+echo "END DEFINE"
```

Comment:
This might be an outdated feature from a previous major release of `NSIMD`. Currently the only important option is the architecture which is encoded in the library file name (e.g. `sse2`, `avx512`) used for linking.

Reply:
Grr. GitHub ate my comment. Here we go again.

Alright, so I did some testing to see what the build system and compiler actually do.

I tested by removing or adding the the `DEFINE` and `CXXFLAGS=` bits of the commit.

I found that:

1. for building `libNSIMD_AVX2.so` it is sufficient to have `-Dsimd_AVX2` set via CMake to get `vmovd` instructions show up in the disassembled output (and `movsd` if one uses `NSIMD_SIMD=CPU`) so that fiddling with CXXFLAGS to build the library and have it use AVX2 is not required (have not tested `FMA`).
2. for buidling client code (ie Cactus code using `nsimd_all.h`) I do seem to need the `DEFINE` as tested using the NSIMD `tutorial.cpp` which switches instructions used for the `mask_storeu` used based on whether `NSIMD_AVX2` is defined or not (`vmovdqa` vs `movslq`). 

The `NSIMD_FMA` and `NSIMD_AVX2` defines are checked for in `nsimd.h` but not defined there based on any build time (of libNSIMD_AVX2.so) options as far as I can tell.

Test code is in a branch rhaas/nsimd-test and the disassembled code in the attached txt files (GitHub limits the allowed extensions).

[test.cc-noAVX2.S.txt](https://github.com/EinsteinToolkit/ExternalLibraries-NSIMD/files/13572421/test.cc-noAVX2.S.txt)
[test.cc-AVX2.S.txt](https://github.com/EinsteinToolkit/ExternalLibraries-NSIMD/files/13572422/test.cc-AVX2.S.txt)

Reply:
Interesting...

Reply:
Might need we look into vectorization in Z4c benchmarks once more :-)